### ISSUE TITLE
[BUG] 2 chrome browsers opened when running a test against selenium 4 grid.

### ISSUE THREAD

**Issue by quldude** [microsoft/playwright#22240]
### System info
- Playwright Version: 1.32.1
- Operating System: Windows 10
- Browser: Chrome
- Other info: Selenium 4 grid - hub and 2 nodes mode in same machine with chromedriver.

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

```js
const config: PlaywrightTestConfig = {
  testDir: '.',
  /* Maximum time one test can run for. */
  timeout: 3600 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 5 * 1000,
  },
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  retries: 1,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [['html'], ['list', { printSteps: true }]],
  globalSetup: require.resolve('./lib/global-setup'),
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 0,
    /* Base URL to use in actions like `await page.goto('/')`. */
    baseURL: process.env.CI ? process.env.TEST_SERVER_URL : 'http://localhost:3000',

    /* Use data-qa as the default test-id attribute */
    testIdAttribute: 'data-qa',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
    screenshot: 'only-on-failure',
    video: 'retain-on-failure',
    headless: false,
    userAgent: `${devices['Desktop Chrome'].userAgent} ${customUserAgentSuffix}`,
    channel: 'chrome',
  },

  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
  outputDir: 'test-results/',
};
```

**Steps**
- Run command `DEBUG=pw:browser* SELENIUM_REMOTE_URL=http://{hub ip}:4444 PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1 npx playwright test specs/tests.spec.ts -g "test name"`.

**Expected**

Only 1 chrome browser opened with test site page..

**Actual**

2 chrome browsers opened. One opens with blank page and the other opens with test site page.

**Comment by pavelfeldman** [comment]
Selenium Grid support in Playwright is highly experimental, not sure what's happening there, will leave open for the others to help if they can.

**Comment by UkrainianHulk** [comment]
Same here.

**Comment by annguyen2805** [comment]
same here. when running playwright with selenium grid 4.10 with playwright appear 2 chrome in Node.

**Comment by egorsoroka8** [comment]
Hi everyone,
I'm experiencing the same problem with a Linux node. I don't have an in-depth understanding of Selenium Grid processes, but if I get correctly, when we attach to the node, WebDriver automatically opens a browser instance. If we use the built-in Playwright page fixture, it also opens a browser, resulting in two browser instances.
 
After some experiments, I managed to avoid the second instance from opening and use the first one. You can create a new project, install the 'playwright' and 'webdriverio' packages, and use the following code:
 
```javascript
import { chromium } from 'playwright';
import { remote } from 'webdriverio';

const grid = {
  ip: 'my_ip', // your selenium hub ip
  port: 4444, // your selenium hub port
};

(async () => {
  // Connect to Selenium Grid via WebDriver
  const browser = await remote({
    capabilities: {
      browserName: 'chrome',
      platformName: 'linux', // set here desired os
    },
    hostname: grid.ip,
    port: grid.port,
    path: '/wd/hub'
  });
  
  // Get browser session id
  const sessionId = browser['sessionId'];
  console.log(`sessionId: ${sessionId}`);
  
  // Get browser end point
  const wsEndPoint = `ws://${grid.ip}:${grid.port}/session/${sessionId}/se/cdp`;
  console.log(`wsEndPoint: ${wsEndPoint}`);

  // Get browser page
  const pwBrowser = await chromium.connectOverCDP(wsEndPoint);
  const context = pwBrowser.contexts()[0];
  const page = context.pages()[0];

  // Run test
  await page.goto('https://google.com');
  await page.waitForTimeout(20_000); // just to have time to check browser in node
  const title = await page.title();
  
  // Log result
  console.log(title);
  
  // Close 
  await pwBrowser.close();
  await browser.deleteSession();
})();
```

There is a possibility that my conclusions are not entirely correct and the situation is a little different, but I think that the truth is somewhere nearby.

@pavelfeldman 
Since Selenium Grid works in experimental mode, I assume that this case will not be processed on the playwright side, but maybe you can suggest another way to get the browser sessionId without installation WebdriverIO? I can see it in logs when running tests through Selenium Grid.

<img width="1146" alt="selenium-playwright-logs" src="https://github.com/user-attachments/assets/96448d27-309c-4325-a7ce-0750227a7453">

Thanks in advance.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
